test/test-test.js

```diff
@@ -233,6 +233,69 @@
             }).call({}, "arg1", {}, done);
         },
 
+        "async promise-based test": {
+            requiresSupportFor: {
+                Promise: typeof Promise !== "undefined"
+            },
+
+            "restores the sandbox when the promise is fulfilled": function (done) {
+                var method = function () {};
+                var object = { method: method };
+                var spy = sinon.spy();
+                sinon.test(function (callback) {
+                    var stub;
+                    var sandbox = this;
+                    return Promise.resolve()
+                        .then(function () {
```

Comment:
Even if you create the `stub`, you don't need to call it and confirm it was called. That is tested elsewhere. At most I would stub it and confirm it was no longer stubbed at the end, but that seems unnecessary when you can confirm `sandbox.restore` was called. I guess it's hard to confirm when it was called though. I'm fine with leaving the stub in, but no need to call it and test that stubs work in this file.